### PR TITLE
[Function](array) support array_range function

### PR DESCRIPTION
**Pull request by zhangstar333**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 40.49 seconds
 load time: 632 seconds
 storage size: 17154875762 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221021123004_clickbench_pr_32521.html

File: be/src/vec/functions/array/function_array_range.cpp [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+    /// Get function name.
+    String get_name() const override { return name; }
+
+    bool is_variadic() const override { return true; }
+
+    bool use_default_implementation_for_nulls() const override { return false; }
+
+    //TODO: now the input have no limit, so it's column or const number
+    //But we maybe could set the step argument is const not column, this could impl the vector_const()
+    bool use_default_implementation_for_constants() const override { return false; }
```
Comment (HappenLee) [inline be/src/vec/functions/array/function_array_range.cpp:49]: the variable should be `true`

File: be/src/vec/functions/array/function_array_range.cpp [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+                       const PaddedPODArray<Int32>& step, NullMap& args_null_map,
+                       PaddedPODArray<Int32>& nested_column,
+                       PaddedPODArray<UInt8>& dest_nested_null_map,
+                       ColumnArray::Offsets64& dest_offsets) {
+        int rows = start.size();
+        for (auto row = 0; row < rows; ++row) {
+            if (args_null_map[row] || start[row] < 0 || end[row] < 0 || step[row] < 0) {
+                nested_column.push_back(0);
+                dest_offsets.push_back(dest_offsets.back() + 1);
+                dest_nested_null_map.push_back(1);
```
Comment (HappenLee) [inline be/src/vec/functions/array/function_array_range.cpp:133]: why here need `push_back 1` ?

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**